Gems/Terrain/Code/Source/TerrainRenderer/TerrainDetailMaterialManager.cpp

```diff
@@ -751,50 +751,56 @@ namespace Terrain
-                                {
-                                    pixels.at(index).m_material1 = aznumeric_cast<uint8_t>(materialId);
-                                    firstWeight = surfaceTagWeight.m_weight;
-                                    // m_blend only needs to be calculated is material 2 is found, otherwise the initial value of 0 is correct.
-                                    isFirstMaterial = false;
-                                }
-                                else
-                                {
-                                    pixels.at(index).m_material2 = aznumeric_cast<uint8_t>(materialId);
-                                    float totalWeight = firstWeight + surfaceTagWeight.m_weight;
-                                    float blendWeight = 1.0f - (firstWeight / totalWeight);
-                                    pixels.at(index).m_blend = aznumeric_cast<uint8_t>(AZStd::round(blendWeight * 255.0f));
-                                    break;
-                                }
+                                pixels.at(index).m_material1 = aznumeric_cast<uint8_t>(materialId);
+                                firstWeight = surfaceTagWeight.m_weight;
+                                // m_blend only needs to be calculated is material 2 is found, otherwise the initial value of 0 is correct.
+                                isFirstMaterial = false;
+                            }
+                            else
+                            {
+                                pixels.at(index).m_material2 = aznumeric_cast<uint8_t>(materialId);
+                                float totalWeight = firstWeight + surfaceTagWeight.m_weight;
+                                float blendWeight = 1.0f - (firstWeight / totalWeight);
+                                pixels.at(index).m_blend = aznumeric_cast<uint8_t>(AZStd::round(blendWeight * 255.0f));
+                                break;
                             }
-                        }
-                        else
-                        {
-                            break; // since the list is ordered, no other materials are in the list with positive weights.
                         }
                     }
-                    ++index;
+                    else
+                    {
+                        break; // since the list is ordered, no other materials are in the list with positive weights.
+                    }
                 }
-            }
+                ++index;
+            };
+            
+            AZ::Vector3 worldMin(quadrantWorldArea.m_min.m_x * DetailTextureScale, quadrantWorldArea.m_min.m_y * DetailTextureScale, 0.0f);
+            AZ::Vector3 worldMax(quadrantWorldArea.m_max.m_x * DetailTextureScale, quadrantWorldArea.m_max.m_y * DetailTextureScale, 0.0f);
+            AZ::Vector2 stepSize(DetailTextureScale);
+            AZ::Aabb region;
+            region.Set(worldMin, worldMax);
```

Comment:
We create an Aabb to replicates the looping implementation. In both cases, the value of x and y coordinates passed to obtain the surface weights will evaluate to the same value.
The looping version : `(quadrantWorldArea.m_min.m_<x/y>  + index) * DetailTextureScale`
The Process region version: `(quadrantWorldArea.m_min.m_<x/y> * DetailTextureScale) + (index * DetailTextureScale)`